### PR TITLE
Update to go 1.18

### PR DESCRIPTION
**Pull request by koba1t**
This PR is a part of https://github.com/kubernetes-sigs/kustomize/pull/4692

Bump all go.mod files and github workflow to go 1.18.

### CODE REVIEW

File: releasing/cloudbuild.yaml [hunk 1]
```diff
@@ -2,53 +2,53 @@
...
-  - 'echo'
-  - 'Cloud build substitution check: '
-  - 'BUILD_ID=$BUILD_ID'
-  - 'PROJECT_ID=$PROJECT_ID'
-  - 'REVISION_ID=$REVISION_ID'
-  - 'REPO_NAME=$REPO_NAME'
-  - 'COMMIT_SHA=$COMMIT_SHA'
-  - 'BRANCH_NAME=$BRANCH_NAME'
-  - 'TAG_NAME=$TAG_NAME'
+  - name: "bash"
```
Comment (natasha41575) [inline releasing/cloudbuild.yaml:5]: Is there a reason you committed this file? This relates to the releasing process and I'd prefer it to not be modified unless it's necessary.

Comment (koba1t) [reply releasing/cloudbuild.yaml]: I think my editor was lint to this yaml file.
I restore it from master.

**Comment by koba1t** [comment]
@natasha41575 
I restored `cloudbuild.yaml`.
Please recheck it?

**Pull request merged**